portal/tasks.py

```diff
@@ -304,7 +304,7 @@ def generate_and_send_summaries(cutoff_days, org_id):
     name_key = SiteSummaryEmail_ATMA.name_key(org=top_org.name)
 
     for user in User.query.filter_by(deleted_id=None).all():
-        if (user.has_role(ROLE.STAFF) and (u'@' in user.email)
+        if (user.has_role(ROLE.STAFF) and user.email and (u'@' in user.email)
```

Comment:
Why not move those filters into the query?

Reply:
* The `user.has_role(ROLE.STAFF)` filter _could_ be moved to the query, but it would make it way more complex (joining three tables, etc). Figured since we already had the check here, it'd be simpler code to just included it here.
* The `(top_org in ot.find_top_level_org(user.organizations))` can't be moved to the query
* The email checks act differently in sqlalchemy vs being called directly; that is, `user.email` does the whole prefix-removal-thing when called directly, but when called in a sqlalchemy query filter, it actually just returns `user._email`. I mentioned it to Paul [in this PR](https://github.com/uwcirg/true_nth_usa_portal/pull/1530#discussion_r146428689) and [in this PT story](https://www.pivotaltracker.com/n/projects/1225464/stories/151111342), but it has something to do with how we (apparently improperly) set up the hybrid property there. In any case, would rather just handle that here for now, until that property gets properly fixed.

Reply:
Actually, thinking more about this - for the `user.email` checks, it shouldn't matter if we're included the prefixes or not (at least, for this initial check that the email address includes '@'; obviously, we don't want the prefixes when we actually pull the `user.email` value to use as the `EmailMessage.recipient`). So you're right @ivan-c , that can probably be moved into the query. I'll make that change now.

Reply:
>user.email
Thanks, forgot that was a method underneath and not actually in the DB

Reply:
ACTUALLY actually, we do still want to check that `user.email is not None`, because in the User model, if `user._email` starts with the no_email prefix (e.g. `user._email == '__no_email__test@gmail.com'`), then `user.email` returns None. So it'd be possible that in the sqlalchemy query, the check to see if `user.email contains '@'` returns True (because it actually checks against `user._email`), but then we'd get errors when we try to actually send the EmaiMessage.

Long story short, gonna keep this as is for now :)